### PR TITLE
[RELAY]take and transpose comp and schd

### PR DESCRIPTION
**Pull request by siju-samuel**
#2051 

Thanks for contributing to TVM!   Please refer to guideline https://docs.tvm.ai/contribute/ for useful information and tips. After the pull request is submitted, please request code reviews from [Reviewers](https://github.com/dmlc/tvm/blob/master/CONTRIBUTORS.md#reviewers).

### CODE REVIEW

**Comment by siju-samuel** [comment]
@jroesch, @MarisaKirisame @tqchen please help to review this PR

File: src/relay/op/tensor/transform.cc [hunk 1]
```diff
@@ -522,7 +529,19 @@ Examples::
...
-.add_type_rel("Take", TakeRel);
+.add_type_rel("Take", TakeRel)
+.set_attr<FTVMCompute>("FTVMCompute", [](const Attrs& attrs,
+                                         const Array<Tensor>& inputs,
+                                         const Type& out_type,
+                                         const Target& target) {
+  const auto* param = attrs.as<TakeAttrs>();
+  CHECK(param != nullptr);
+  if (!param->axis.defined()) {
+    return Array<Tensor>{topi::take(inputs[0], inputs[1]) };
```
Comment (MarisaKirisame) [inline src/relay/op/tensor/transform.cc]: why {} constructor?

File: src/relay/op/op_common.h [hunk 2]
```diff
@@ -74,6 +74,22 @@ std::vector<T> AsVector(const Array<T> &array) {
...
 
+/*
+ * \brief Helper function to convert Integer array to TVM array. useful for
+ * passing data from Relay structures to TOPI ops.
+ *
+ * \param shape The shape to convert
+ *
+ * \return An Array of Expr
+ */
+inline tvm::Array<tvm::Expr> IntegerArrayToExprArray(const Array<Integer> &shape) {
```
Comment (tqchen) [inline src/relay/op/op_common.h]: We should not have this, instead we should change the topi API to accept Integer instead

File: topi/include/topi/transform.h [hunk 3]
```diff
@@ -86,11 +86,11 @@ inline Tensor expand_dims(const Tensor& x,
 * \return A Tensor whose op member is the transpose operation
 */
 inline Tensor transpose(const Tensor& x,
-                        Array<Expr> axes,
+                        Array<Integer> axes,
                         std::string name = "tensor",
                         std::string tag = kInjective) {
   if (axes.size() == 0) {
```
Comment (tqchen) [inline topi/include/topi/transform.h]: !axes.defined() || axes.size() == 0

File: topi/include/topi/detail/constant_utils.h [hunk 4]
```diff
@@ -86,6 +86,26 @@ inline std::vector<int64_t> GetConstInt64Values(Array<Expr> exprs, const std::st
...
+/*!
+ * \brief Get the value of all the constant integer expressions in the given array
+ *
+ * \param values The array of integers to get the values of
+ * \param var_name The name to be used when logging an error in the event that any
+ * of the expressions are not constant integers.
+ *
+ * \return A vector of the integer values
+ */
+inline std::vector<int> GetConstIntValues(const Array<Integer>& values,
```
Comment (tqchen) [inline topi/include/topi/detail/constant_utils.h]: Likely we don't need this, as it is already easy to get int from values, we should directly operate on Array<Integer> and avoid creating a new vector<int>

File: topi/include/topi/transform.h [hunk 5]
```diff
@@ -86,32 +86,33 @@ inline Tensor expand_dims(const Tensor& x,
...
   }
 
-  auto axes_val = GetConstIntValues(axes, "axes");
-  for (size_t i = 0; i < axes_val.size(); ++i) {
-    int axis = axes_val[i];
-    if (axes_val[i] < 0) {
-      axes_val[i] = static_cast<int>(x->shape.size()) + axes_val[i];
+  std::vector<int> axes_val;
+  for (size_t i = 0; i < axes.size(); ++i) {
+    int axis = GetConstInt(axes[i]);
```
Comment (tqchen) [inline topi/include/topi/transform.h]: int axis = static_cast<int>(axes[i]->value);

File: topi/include/topi/transform.h [hunk 6]
```diff
@@ -86,32 +86,33 @@ inline Tensor expand_dims(const Tensor& x,
...
       << static_cast<int>(x->shape.size()) << "-dimensional input tensor";
-
-    CHECK(1 == std::count(std::begin(axes_val), std::end(axes_val), axes_val[i]))
-      << "repeated axis in transpose";
+    axes_val.push_back(new_axis);
   }
 
   Array<Expr> new_shape;
   for (size_t i = 0; i < axes_val.size(); ++i) {
+    CHECK(1 == std::count(std::begin(axes_val), std::end(axes_val), axes_val[i]))
```
Comment (tqchen) [inline topi/include/topi/transform.h]: std::count has linear complexity, makin O(n^2) consider create a bool flag of dimension instead

**Comment by tqchen** [comment]
Thanks @siju-samuel @MarisaKirisame , this is merged

**Pull request merged**